File: app/src/main/java/com/kidozh/discuzhub/results/AddCheckResult.java
package com.kidozh.discuzhub.results;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.kidozh.discuzhub.entities.bbsInformation;
import com.kidozh.discuzhub.utilities.OneZeroBooleanJsonDeserializer;

@JsonIgnoreProperties(ignoreUnknown=true)
public class AddCheckResult {
    @JsonProperty("discuzversion")
    public String discuz_version;
    public String charset;
    @JsonProperty("version")
    public String apiVersion;
    @JsonProperty("pluginversion")
    public String pluginVersion;
    @JsonProperty("regname")
    public String registerName;
    @JsonFormat(shape = JsonFormat.Shape.STRING)
    @JsonDeserialize(using = OneZeroBooleanJsonDeserializer.class)
    @JsonProperty("qqconnect")
    public boolean qqConnect;
    @JsonFormat(shape = JsonFormat.Shape.STRING)
    @JsonDeserialize(using = OneZeroBooleanJsonDeserializer.class)
    @JsonProperty("wsqqqconnect")
    public boolean wsqQQConnect;
    @JsonFormat(shape = JsonFormat.Shape.STRING)
    @JsonDeserialize(using = OneZeroBooleanJsonDeserializer.class)
    @JsonProperty("wsqhideregister")
    public boolean wsqHideRegister;
    @JsonProperty("sitename")
    public String siteName;
    @JsonProperty("mysiteid")
    public String mySiteId;
    @JsonProperty("ucenterurl")
    public String uCenterURL;
    @JsonProperty("defaultfid")
    public String defaultFid;
    @JsonProperty("totalposts")
    public int totalPosts;
    @JsonProperty("totalmembers")
    public int totalMembers;
    @JsonProperty("testcookie")
    public String testCookie;

    public bbsInformation toBBSInformation(String baseUrl){
        return new bbsInformation(baseUrl,siteName,discuz_version,charset,apiVersion,pluginVersion,
                String.valueOf(totalPosts),String.valueOf(totalMembers),String.valueOf(mySiteId),
                defaultFid,uCenterURL,registerName,"",wsqHideRegister,qqConnect
                );
    }
}

File: app/src/main/java/com/kidozh/discuzhub/results/AddCheckResult.kt
package com.kidozh.discuzhub.results

import com.fasterxml.jackson.annotation.JsonFormat
import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty
import com.fasterxml.jackson.databind.annotation.JsonDeserialize
import com.kidozh.discuzhub.entities.Discuz
import com.kidozh.discuzhub.utilities.OneZeroBooleanJsonDeserializer


@JsonIgnoreProperties(ignoreUnknown = true)
class AddCheckResult() {
    @JsonProperty("discuzversion")
    var discuz_version: String = ""
    var charset: String = ""

    @JsonProperty("version")
    var apiVersion = 0

    @JsonProperty("pluginversion")
    var pluginVersion: String=""

    @JsonProperty("regname")
    var registerName: String = ""

    @JsonFormat(shape = JsonFormat.Shape.STRING)
    @JsonDeserialize(using = OneZeroBooleanJsonDeserializer::class)
    @JsonProperty("qqconnect")
    var qqConnect = false

    @JsonFormat(shape = JsonFormat.Shape.STRING)
    @JsonDeserialize(using = OneZeroBooleanJsonDeserializer::class)
    @JsonProperty("wsqqqconnect")
    var wsqQQConnect = false

    @JsonFormat(shape = JsonFormat.Shape.STRING)
    @JsonDeserialize(using = OneZeroBooleanJsonDeserializer::class)
    @JsonProperty("wsqhideregister")
    var wsqHideRegister = false


    @JsonProperty("sitename")
    var siteName: String = ""

    @JsonProperty("mysiteid")
    var mySiteId: String = ""

    @JsonProperty("ucenterurl")
    var uCenterURL: String = ""

    @JsonProperty("defaultfid")
    var defaultFid = 0

    @JsonProperty("totalposts")
    var totalPosts = 0L

    @JsonProperty("totalmembers")
    var totalMembers = 0L

    @JsonProperty("testcookie")
    var testCookie: String? = null
    fun toBBSInformation(baseUrl: String): Discuz {
        return Discuz(baseUrl, siteName, discuz_version, charset, apiVersion, pluginVersion, totalPosts, totalMembers, mySiteId,
                defaultFid, uCenterURL, registerName, "", wsqHideRegister, qqConnect
        )
    }
}